GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrMonoscopicViewManager.java

```diff
@@ -126,32 +140,129 @@
+            BufferedReader reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+
+            if(reader.readLine().equals("1")){
+                isVulkanInstance = true;
+                mRenderTarget[0] = null;
+                gvrActivity.getAppSettings().getEyeBufferParams().setResolutionWidth(mViewportWidth);
+                gvrActivity.getAppSettings().getEyeBufferParams().setResolutionHeight(mViewportHeight);
+                gvrActivity.getAppSettings().getEyeBufferParams().setMultiSamples(sampleCount);
+                vulkanSurfaceView = new SurfaceView(mActivity);
+
+                vulkanSurfaceView.getHolder().addCallback(new SurfaceHolder.Callback() {
+                    @Override
+                    public void surfaceCreated(SurfaceHolder surfaceHolder) {
+                        Log.d("Vulkan", "surfaceCreated");
+                        workerObj = new Worker();
+                        vulkanDrawThread = new Thread(workerObj, "vulkanThread");
+                        if(!initialized) {
+                            mRenderBundle = makeRenderBundle();
+                            final GVRScene scene = null == mMainScene ? new GVRScene(OvrMonoscopicViewManager.this) : mMainScene;
+                            mMainScene = scene;
+                            NativeScene.setMainScene(scene.getNative());
+                            getActivity().setCameraRig(scene.getMainCameraRig());
+                            mInputManager.setScene(scene);
+                            mRotationSensor.onResume();
+                            long vulkanCoreObj;
+                            vulkanCoreObj = NativeVulkanCore.getInstance(vulkanSurfaceView.getHolder().getSurface());
+                            Thread currentThread = Thread.currentThread();
+
+                            // Reduce contention with other Android processes
+                            currentThread.setPriority(Thread.MAX_PRIORITY);
+
+                            if (vulkanCoreObj != 0) {
+                                Log.i("Vulkan", "Vulkan Instance On surface created at Vulkan Java Side");
+                            } else {
+                                Log.i("Vulkan", "Error : On surface  No Instance created at Vulkan Java Side");
+                            }
+
+                            initialized = true;
+                        }
+                        else{
+                            NativeVulkanCore.resetTheInstance();
+                            NativeVulkanCore.getInstance(vulkanSurfaceView.getHolder().getSurface());
+                        }
+                    }
+
+                    @Override
+                    public void surfaceChanged(SurfaceHolder surfaceHolder, int i, int i1, int i2) {
+                        Log.d("Vulkan", "surfaceChanged");
+                        vulkanDrawThread.start();
```

Comment:
surfaceChanged can be called multiple times. Thread.start() can't be called more than once per instance.